// @azure/arm-compute/esm/operations/cloudServiceOperatingSystems.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/cloudServiceOperatingSystemsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a CloudServiceOperatingSystems. */
var CloudServiceOperatingSystems = /** @class */ (function () {
    /**
     * Create a CloudServiceOperatingSystems.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function CloudServiceOperatingSystems(client) {
        this.client = client;
    }
    CloudServiceOperatingSystems.prototype.getOSVersion = function (location, osVersionName, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            osVersionName: osVersionName,
            options: options
        }, getOSVersionOperationSpec, callback);
    };
    CloudServiceOperatingSystems.prototype.listOSVersions = function (location, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            options: options
        }, listOSVersionsOperationSpec, callback);
    };
    CloudServiceOperatingSystems.prototype.getOSFamily = function (location, osFamilyName, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            osFamilyName: osFamilyName,
            options: options
        }, getOSFamilyOperationSpec, callback);
    };
    CloudServiceOperatingSystems.prototype.listOSFamilies = function (location, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            options: options
        }, listOSFamiliesOperationSpec, callback);
    };
    CloudServiceOperatingSystems.prototype.listOSVersionsNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listOSVersionsNextOperationSpec, callback);
    };
    CloudServiceOperatingSystems.prototype.listOSFamiliesNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listOSFamiliesNextOperationSpec, callback);
    };
    return CloudServiceOperatingSystems;
}());
export { CloudServiceOperatingSystems };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOSVersionOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/cloudServiceOsVersions/{osVersionName}",
    urlParameters: [
        Parameters.location0,
        Parameters.osVersionName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OSVersion
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOSVersionsOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/cloudServiceOsVersions",
    urlParameters: [
        Parameters.location0,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OSVersionListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var getOSFamilyOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/cloudServiceOsFamilies/{osFamilyName}",
    urlParameters: [
        Parameters.location0,
        Parameters.osFamilyName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OSFamily
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOSFamiliesOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/cloudServiceOsFamilies",
    urlParameters: [
        Parameters.location0,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OSFamilyListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOSVersionsNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OSVersionListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOSFamiliesNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion4
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.OSFamilyListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
