// @azure/arm-compute/esm/operations/dedicatedHosts.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/dedicatedHostsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a DedicatedHosts. */
var DedicatedHosts = /** @class */ (function () {
    /**
     * Create a DedicatedHosts.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function DedicatedHosts(client) {
        this.client = client;
    }
    /**
     * Create or update a dedicated host .
     * @param resourceGroupName The name of the resource group.
     * @param hostGroupName The name of the dedicated host group.
     * @param hostName The name of the dedicated host .
     * @param parameters Parameters supplied to the Create Dedicated Host.
     * @param [options] The optional parameters
     * @returns Promise<Models.DedicatedHostsCreateOrUpdateResponse>
     */
    DedicatedHosts.prototype.createOrUpdate = function (resourceGroupName, hostGroupName, hostName, parameters, options) {
        return this.beginCreateOrUpdate(resourceGroupName, hostGroupName, hostName, parameters, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Update an dedicated host .
     * @param resourceGroupName The name of the resource group.
     * @param hostGroupName The name of the dedicated host group.
     * @param hostName The name of the dedicated host .
     * @param parameters Parameters supplied to the Update Dedicated Host operation.
     * @param [options] The optional parameters
     * @returns Promise<Models.DedicatedHostsUpdateResponse>
     */
    DedicatedHosts.prototype.update = function (resourceGroupName, hostGroupName, hostName, parameters, options) {
        return this.beginUpdate(resourceGroupName, hostGroupName, hostName, parameters, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Delete a dedicated host.
     * @param resourceGroupName The name of the resource group.
     * @param hostGroupName The name of the dedicated host group.
     * @param hostName The name of the dedicated host.
     * @param [options] The optional parameters
     * @returns Promise<msRest.RestResponse>
     */
    DedicatedHosts.prototype.deleteMethod = function (resourceGroupName, hostGroupName, hostName, options) {
        return this.beginDeleteMethod(resourceGroupName, hostGroupName, hostName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    DedicatedHosts.prototype.get = function (resourceGroupName, hostGroupName, hostName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            hostGroupName: hostGroupName,
            hostName: hostName,
            options: options
        }, getOperationSpec, callback);
    };
    DedicatedHosts.prototype.listByHostGroup = function (resourceGroupName, hostGroupName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            hostGroupName: hostGroupName,
            options: options
        }, listByHostGroupOperationSpec, callback);
    };
    /**
     * Create or update a dedicated host .
     * @param resourceGroupName The name of the resource group.
     * @param hostGroupName The name of the dedicated host group.
     * @param hostName The name of the dedicated host .
     * @param parameters Parameters supplied to the Create Dedicated Host.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    DedicatedHosts.prototype.beginCreateOrUpdate = function (resourceGroupName, hostGroupName, hostName, parameters, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            hostGroupName: hostGroupName,
            hostName: hostName,
            parameters: parameters,
            options: options
        }, beginCreateOrUpdateOperationSpec, options);
    };
    /**
     * Update an dedicated host .
     * @param resourceGroupName The name of the resource group.
     * @param hostGroupName The name of the dedicated host group.
     * @param hostName The name of the dedicated host .
     * @param parameters Parameters supplied to the Update Dedicated Host operation.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    DedicatedHosts.prototype.beginUpdate = function (resourceGroupName, hostGroupName, hostName, parameters, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            hostGroupName: hostGroupName,
            hostName: hostName,
            parameters: parameters,
            options: options
        }, beginUpdateOperationSpec, options);
    };
    /**
     * Delete a dedicated host.
     * @param resourceGroupName The name of the resource group.
     * @param hostGroupName The name of the dedicated host group.
     * @param hostName The name of the dedicated host.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    DedicatedHosts.prototype.beginDeleteMethod = function (resourceGroupName, hostGroupName, hostName, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            hostGroupName: hostGroupName,
            hostName: hostName,
            options: options
        }, beginDeleteMethodOperationSpec, options);
    };
    DedicatedHosts.prototype.listByHostGroupNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByHostGroupNextOperationSpec, callback);
    };
    return DedicatedHosts;
}());
export { DedicatedHosts };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}/hosts/{hostName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.hostGroupName,
        Parameters.hostName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.expand1,
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DedicatedHost
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByHostGroupOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}/hosts",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.hostGroupName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DedicatedHostListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginCreateOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}/hosts/{hostName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.hostGroupName,
        Parameters.hostName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.DedicatedHost), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.DedicatedHost
        },
        201: {
            bodyMapper: Mappers.DedicatedHost
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginUpdateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}/hosts/{hostName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.hostGroupName,
        Parameters.hostName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.DedicatedHostUpdate), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.DedicatedHost
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginDeleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/hostGroups/{hostGroupName}/hosts/{hostName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.hostGroupName,
        Parameters.hostName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByHostGroupNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DedicatedHostListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
