// @azure/arm-compute/esm/operations/disks.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/disksMappers";
import * as Parameters from "../models/parameters";
/** Class representing a Disks. */
var Disks = /** @class */ (function () {
    /**
     * Create a Disks.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function Disks(client) {
        this.client = client;
    }
    /**
     * Creates or updates a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param disk Disk object supplied in the body of the Put disk operation.
     * @param [options] The optional parameters
     * @returns Promise<Models.DisksCreateOrUpdateResponse>
     */
    Disks.prototype.createOrUpdate = function (resourceGroupName, diskName, disk, options) {
        return this.beginCreateOrUpdate(resourceGroupName, diskName, disk, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Updates (patches) a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param disk Disk object supplied in the body of the Patch disk operation.
     * @param [options] The optional parameters
     * @returns Promise<Models.DisksUpdateResponse>
     */
    Disks.prototype.update = function (resourceGroupName, diskName, disk, options) {
        return this.beginUpdate(resourceGroupName, diskName, disk, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    Disks.prototype.get = function (resourceGroupName, diskName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            diskName: diskName,
            options: options
        }, getOperationSpec, callback);
    };
    /**
     * Deletes a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param [options] The optional parameters
     * @returns Promise<msRest.RestResponse>
     */
    Disks.prototype.deleteMethod = function (resourceGroupName, diskName, options) {
        return this.beginDeleteMethod(resourceGroupName, diskName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    Disks.prototype.listByResourceGroup = function (resourceGroupName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            options: options
        }, listByResourceGroupOperationSpec, callback);
    };
    Disks.prototype.list = function (options, callback) {
        return this.client.sendOperationRequest({
            options: options
        }, listOperationSpec, callback);
    };
    /**
     * Grants access to a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param grantAccessData Access data object supplied in the body of the get disk access operation.
     * @param [options] The optional parameters
     * @returns Promise<Models.DisksGrantAccessResponse>
     */
    Disks.prototype.grantAccess = function (resourceGroupName, diskName, grantAccessData, options) {
        return this.beginGrantAccess(resourceGroupName, diskName, grantAccessData, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Revokes access to a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param [options] The optional parameters
     * @returns Promise<msRest.RestResponse>
     */
    Disks.prototype.revokeAccess = function (resourceGroupName, diskName, options) {
        return this.beginRevokeAccess(resourceGroupName, diskName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Creates or updates a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param disk Disk object supplied in the body of the Put disk operation.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Disks.prototype.beginCreateOrUpdate = function (resourceGroupName, diskName, disk, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            diskName: diskName,
            disk: disk,
            options: options
        }, beginCreateOrUpdateOperationSpec, options);
    };
    /**
     * Updates (patches) a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param disk Disk object supplied in the body of the Patch disk operation.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Disks.prototype.beginUpdate = function (resourceGroupName, diskName, disk, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            diskName: diskName,
            disk: disk,
            options: options
        }, beginUpdateOperationSpec, options);
    };
    /**
     * Deletes a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Disks.prototype.beginDeleteMethod = function (resourceGroupName, diskName, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            diskName: diskName,
            options: options
        }, beginDeleteMethodOperationSpec, options);
    };
    /**
     * Grants access to a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param grantAccessData Access data object supplied in the body of the get disk access operation.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Disks.prototype.beginGrantAccess = function (resourceGroupName, diskName, grantAccessData, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            diskName: diskName,
            grantAccessData: grantAccessData,
            options: options
        }, beginGrantAccessOperationSpec, options);
    };
    /**
     * Revokes access to a disk.
     * @param resourceGroupName The name of the resource group.
     * @param diskName The name of the managed disk that is being created. The name can't be changed
     * after the disk is created. Supported characters for the name are a-z, A-Z, 0-9 and _. The
     * maximum name length is 80 characters.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    Disks.prototype.beginRevokeAccess = function (resourceGroupName, diskName, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            diskName: diskName,
            options: options
        }, beginRevokeAccessOperationSpec, options);
    };
    Disks.prototype.listByResourceGroupNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByResourceGroupNextOperationSpec, callback);
    };
    Disks.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return Disks;
}());
export { Disks };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.diskName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.Disk
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByResourceGroupOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DiskList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/disks",
    urlParameters: [
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DiskList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginCreateOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.diskName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "disk",
        mapper: __assign(__assign({}, Mappers.Disk), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.Disk
        },
        202: {
            bodyMapper: Mappers.Disk
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginUpdateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.diskName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "disk",
        mapper: __assign(__assign({}, Mappers.DiskUpdate), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.Disk
        },
        202: {
            bodyMapper: Mappers.Disk
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginDeleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.diskName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginGrantAccessOperationSpec = {
    httpMethod: "POST",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}/beginGetAccess",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.diskName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "grantAccessData",
        mapper: __assign(__assign({}, Mappers.GrantAccessData), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.AccessUri
        },
        202: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginRevokeAccessOperationSpec = {
    httpMethod: "POST",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/disks/{diskName}/endGetAccess",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.diskName
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByResourceGroupNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DiskList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion2
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.DiskList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
