// @azure/arm-compute/esm/operations/galleryImages.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/galleryImagesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a GalleryImages. */
var GalleryImages = /** @class */ (function () {
    /**
     * Create a GalleryImages.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function GalleryImages(client) {
        this.client = client;
    }
    /**
     * Create or update a gallery image definition.
     * @param resourceGroupName The name of the resource group.
     * @param galleryName The name of the Shared Image Gallery in which the Image Definition is to be
     * created.
     * @param galleryImageName The name of the gallery image definition to be created or updated. The
     * allowed characters are alphabets and numbers with dots, dashes, and periods allowed in the
     * middle. The maximum length is 80 characters.
     * @param galleryImage Parameters supplied to the create or update gallery image operation.
     * @param [options] The optional parameters
     * @returns Promise<Models.GalleryImagesCreateOrUpdateResponse>
     */
    GalleryImages.prototype.createOrUpdate = function (resourceGroupName, galleryName, galleryImageName, galleryImage, options) {
        return this.beginCreateOrUpdate(resourceGroupName, galleryName, galleryImageName, galleryImage, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Update a gallery image definition.
     * @param resourceGroupName The name of the resource group.
     * @param galleryName The name of the Shared Image Gallery in which the Image Definition is to be
     * updated.
     * @param galleryImageName The name of the gallery image definition to be updated. The allowed
     * characters are alphabets and numbers with dots, dashes, and periods allowed in the middle. The
     * maximum length is 80 characters.
     * @param galleryImage Parameters supplied to the update gallery image operation.
     * @param [options] The optional parameters
     * @returns Promise<Models.GalleryImagesUpdateResponse>
     */
    GalleryImages.prototype.update = function (resourceGroupName, galleryName, galleryImageName, galleryImage, options) {
        return this.beginUpdate(resourceGroupName, galleryName, galleryImageName, galleryImage, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    GalleryImages.prototype.get = function (resourceGroupName, galleryName, galleryImageName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            galleryName: galleryName,
            galleryImageName: galleryImageName,
            options: options
        }, getOperationSpec, callback);
    };
    /**
     * Delete a gallery image.
     * @param resourceGroupName The name of the resource group.
     * @param galleryName The name of the Shared Image Gallery in which the Image Definition is to be
     * deleted.
     * @param galleryImageName The name of the gallery image definition to be deleted.
     * @param [options] The optional parameters
     * @returns Promise<msRest.RestResponse>
     */
    GalleryImages.prototype.deleteMethod = function (resourceGroupName, galleryName, galleryImageName, options) {
        return this.beginDeleteMethod(resourceGroupName, galleryName, galleryImageName, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    GalleryImages.prototype.listByGallery = function (resourceGroupName, galleryName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            galleryName: galleryName,
            options: options
        }, listByGalleryOperationSpec, callback);
    };
    /**
     * Create or update a gallery image definition.
     * @param resourceGroupName The name of the resource group.
     * @param galleryName The name of the Shared Image Gallery in which the Image Definition is to be
     * created.
     * @param galleryImageName The name of the gallery image definition to be created or updated. The
     * allowed characters are alphabets and numbers with dots, dashes, and periods allowed in the
     * middle. The maximum length is 80 characters.
     * @param galleryImage Parameters supplied to the create or update gallery image operation.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    GalleryImages.prototype.beginCreateOrUpdate = function (resourceGroupName, galleryName, galleryImageName, galleryImage, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            galleryName: galleryName,
            galleryImageName: galleryImageName,
            galleryImage: galleryImage,
            options: options
        }, beginCreateOrUpdateOperationSpec, options);
    };
    /**
     * Update a gallery image definition.
     * @param resourceGroupName The name of the resource group.
     * @param galleryName The name of the Shared Image Gallery in which the Image Definition is to be
     * updated.
     * @param galleryImageName The name of the gallery image definition to be updated. The allowed
     * characters are alphabets and numbers with dots, dashes, and periods allowed in the middle. The
     * maximum length is 80 characters.
     * @param galleryImage Parameters supplied to the update gallery image operation.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    GalleryImages.prototype.beginUpdate = function (resourceGroupName, galleryName, galleryImageName, galleryImage, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            galleryName: galleryName,
            galleryImageName: galleryImageName,
            galleryImage: galleryImage,
            options: options
        }, beginUpdateOperationSpec, options);
    };
    /**
     * Delete a gallery image.
     * @param resourceGroupName The name of the resource group.
     * @param galleryName The name of the Shared Image Gallery in which the Image Definition is to be
     * deleted.
     * @param galleryImageName The name of the gallery image definition to be deleted.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    GalleryImages.prototype.beginDeleteMethod = function (resourceGroupName, galleryName, galleryImageName, options) {
        return this.client.sendLRORequest({
            resourceGroupName: resourceGroupName,
            galleryName: galleryName,
            galleryImageName: galleryImageName,
            options: options
        }, beginDeleteMethodOperationSpec, options);
    };
    GalleryImages.prototype.listByGalleryNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByGalleryNextOperationSpec, callback);
    };
    return GalleryImages;
}());
export { GalleryImages };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImage
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByGalleryOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginCreateOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "galleryImage",
        mapper: __assign(__assign({}, Mappers.GalleryImage), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImage
        },
        201: {
            bodyMapper: Mappers.GalleryImage
        },
        202: {
            bodyMapper: Mappers.GalleryImage
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginUpdateOperationSpec = {
    httpMethod: "PATCH",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "galleryImage",
        mapper: __assign(__assign({}, Mappers.GalleryImageUpdate), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImage
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginDeleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/images/{galleryImageName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.galleryName,
        Parameters.galleryImageName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        202: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByGalleryNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.GalleryImageList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
