// @azure/arm-compute/esm/operations/logAnalytics.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/logAnalyticsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a LogAnalytics. */
var LogAnalytics = /** @class */ (function () {
    /**
     * Create a LogAnalytics.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function LogAnalytics(client) {
        this.client = client;
    }
    /**
     * Export logs that show Api requests made by this subscription in the given time window to show
     * throttling activities.
     * @param parameters Parameters supplied to the LogAnalytics getRequestRateByInterval Api.
     * @param location The location upon which virtual-machine-sizes is queried.
     * @param [options] The optional parameters
     * @returns Promise<Models.LogAnalyticsExportRequestRateByIntervalResponse>
     */
    LogAnalytics.prototype.exportRequestRateByInterval = function (parameters, location, options) {
        return this.beginExportRequestRateByInterval(parameters, location, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Export logs that show total throttled Api requests for this subscription in the given time
     * window.
     * @param parameters Parameters supplied to the LogAnalytics getThrottledRequests Api.
     * @param location The location upon which virtual-machine-sizes is queried.
     * @param [options] The optional parameters
     * @returns Promise<Models.LogAnalyticsExportThrottledRequestsResponse>
     */
    LogAnalytics.prototype.exportThrottledRequests = function (parameters, location, options) {
        return this.beginExportThrottledRequests(parameters, location, options)
            .then(function (lroPoller) { return lroPoller.pollUntilFinished(); });
    };
    /**
     * Export logs that show Api requests made by this subscription in the given time window to show
     * throttling activities.
     * @param parameters Parameters supplied to the LogAnalytics getRequestRateByInterval Api.
     * @param location The location upon which virtual-machine-sizes is queried.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    LogAnalytics.prototype.beginExportRequestRateByInterval = function (parameters, location, options) {
        return this.client.sendLRORequest({
            parameters: parameters,
            location: location,
            options: options
        }, beginExportRequestRateByIntervalOperationSpec, options);
    };
    /**
     * Export logs that show total throttled Api requests for this subscription in the given time
     * window.
     * @param parameters Parameters supplied to the LogAnalytics getThrottledRequests Api.
     * @param location The location upon which virtual-machine-sizes is queried.
     * @param [options] The optional parameters
     * @returns Promise<msRestAzure.LROPoller>
     */
    LogAnalytics.prototype.beginExportThrottledRequests = function (parameters, location, options) {
        return this.client.sendLRORequest({
            parameters: parameters,
            location: location,
            options: options
        }, beginExportThrottledRequestsOperationSpec, options);
    };
    return LogAnalytics;
}());
export { LogAnalytics };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var beginExportRequestRateByIntervalOperationSpec = {
    httpMethod: "POST",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/logAnalytics/apiAccess/getRequestRateByInterval",
    urlParameters: [
        Parameters.location1,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.RequestRateByIntervalInput), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.LogAnalyticsOperationResult
        },
        202: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var beginExportThrottledRequestsOperationSpec = {
    httpMethod: "POST",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/logAnalytics/apiAccess/getThrottledRequests",
    urlParameters: [
        Parameters.location1,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.ThrottledRequestsInput), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.LogAnalyticsOperationResult
        },
        202: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
