// @azure/arm-compute/esm/operations/sharedGalleryImageVersions.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/sharedGalleryImageVersionsMappers";
import * as Parameters from "../models/parameters";
/** Class representing a SharedGalleryImageVersions. */
var SharedGalleryImageVersions = /** @class */ (function () {
    /**
     * Create a SharedGalleryImageVersions.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function SharedGalleryImageVersions(client) {
        this.client = client;
    }
    SharedGalleryImageVersions.prototype.list = function (location, galleryUniqueName, galleryImageName, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            galleryUniqueName: galleryUniqueName,
            galleryImageName: galleryImageName,
            options: options
        }, listOperationSpec, callback);
    };
    SharedGalleryImageVersions.prototype.get = function (location, galleryUniqueName, galleryImageName, galleryImageVersionName, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            galleryUniqueName: galleryUniqueName,
            galleryImageName: galleryImageName,
            galleryImageVersionName: galleryImageVersionName,
            options: options
        }, getOperationSpec, callback);
    };
    SharedGalleryImageVersions.prototype.listNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listNextOperationSpec, callback);
    };
    return SharedGalleryImageVersions;
}());
export { SharedGalleryImageVersions };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images/{galleryImageName}/versions",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.location0,
        Parameters.galleryUniqueName,
        Parameters.galleryImageName
    ],
    queryParameters: [
        Parameters.apiVersion3,
        Parameters.sharedTo
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SharedGalleryImageVersionList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images/{galleryImageName}/versions/{galleryImageVersionName}",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.location0,
        Parameters.galleryUniqueName,
        Parameters.galleryImageName,
        Parameters.galleryImageVersionName
    ],
    queryParameters: [
        Parameters.apiVersion3
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SharedGalleryImageVersion
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    queryParameters: [
        Parameters.apiVersion3,
        Parameters.sharedTo
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.SharedGalleryImageVersionList
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
