// @azure/arm-compute/esm/operations/virtualMachineImagesEdgeZone.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/virtualMachineImagesEdgeZoneMappers";
import * as Parameters from "../models/parameters";
/** Class representing a VirtualMachineImagesEdgeZone. */
var VirtualMachineImagesEdgeZone = /** @class */ (function () {
    /**
     * Create a VirtualMachineImagesEdgeZone.
     * @param {ComputeManagementClientContext} client Reference to the service client.
     */
    function VirtualMachineImagesEdgeZone(client) {
        this.client = client;
    }
    VirtualMachineImagesEdgeZone.prototype.get = function (location, edgeZone, publisherName, offer, skus, version, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            edgeZone: edgeZone,
            publisherName: publisherName,
            offer: offer,
            skus: skus,
            version: version,
            options: options
        }, getOperationSpec, callback);
    };
    VirtualMachineImagesEdgeZone.prototype.list = function (location, edgeZone, publisherName, offer, skus, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            edgeZone: edgeZone,
            publisherName: publisherName,
            offer: offer,
            skus: skus,
            options: options
        }, listOperationSpec, callback);
    };
    VirtualMachineImagesEdgeZone.prototype.listOffers = function (location, edgeZone, publisherName, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            edgeZone: edgeZone,
            publisherName: publisherName,
            options: options
        }, listOffersOperationSpec, callback);
    };
    VirtualMachineImagesEdgeZone.prototype.listPublishers = function (location, edgeZone, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            edgeZone: edgeZone,
            options: options
        }, listPublishersOperationSpec, callback);
    };
    VirtualMachineImagesEdgeZone.prototype.listSkus = function (location, edgeZone, publisherName, offer, options, callback) {
        return this.client.sendOperationRequest({
            location: location,
            edgeZone: edgeZone,
            publisherName: publisherName,
            offer: offer,
            options: options
        }, listSkusOperationSpec, callback);
    };
    return VirtualMachineImagesEdgeZone;
}());
export { VirtualMachineImagesEdgeZone };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/edgeZones/{edgeZone}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions/{version}",
    urlParameters: [
        Parameters.location0,
        Parameters.edgeZone,
        Parameters.publisherName,
        Parameters.offer,
        Parameters.skus,
        Parameters.version,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VirtualMachineImage
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/edgeZones/{edgeZone}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions",
    urlParameters: [
        Parameters.location0,
        Parameters.edgeZone,
        Parameters.publisherName,
        Parameters.offer,
        Parameters.skus,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.expand0,
        Parameters.top,
        Parameters.orderby,
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualMachineImageResource"
                        }
                    }
                }
            }
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listOffersOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/edgeZones/{edgeZone}/publishers/{publisherName}/artifacttypes/vmimage/offers",
    urlParameters: [
        Parameters.location0,
        Parameters.edgeZone,
        Parameters.publisherName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualMachineImageResource"
                        }
                    }
                }
            }
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listPublishersOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/edgeZones/{edgeZone}/publishers",
    urlParameters: [
        Parameters.location0,
        Parameters.edgeZone,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualMachineImageResource"
                        }
                    }
                }
            }
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listSkusOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/edgeZones/{edgeZone}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus",
    urlParameters: [
        Parameters.location0,
        Parameters.edgeZone,
        Parameters.publisherName,
        Parameters.offer,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion0
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VirtualMachineImageResource"
                        }
                    }
                }
            }
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
