// @azure/arm-compute/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var apiVersion0 = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "api-version",
        defaultValue: '2021-04-01',
        type: {
            name: "String"
        }
    }
};
export var apiVersion1 = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "api-version",
        defaultValue: '2019-04-01',
        type: {
            name: "String"
        }
    }
};
export var apiVersion2 = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "api-version",
        defaultValue: '2020-12-01',
        type: {
            name: "String"
        }
    }
};
export var apiVersion3 = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "api-version",
        defaultValue: '2020-09-30',
        type: {
            name: "String"
        }
    }
};
export var apiVersion4 = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "api-version",
        defaultValue: '2021-03-01',
        type: {
            name: "String"
        }
    }
};
export var availabilitySetName = {
    parameterPath: "availabilitySetName",
    mapper: {
        required: true,
        serializedName: "availabilitySetName",
        type: {
            name: "String"
        }
    }
};
export var capacityReservationGroupName = {
    parameterPath: "capacityReservationGroupName",
    mapper: {
        required: true,
        serializedName: "capacityReservationGroupName",
        type: {
            name: "String"
        }
    }
};
export var capacityReservationName = {
    parameterPath: "capacityReservationName",
    mapper: {
        required: true,
        serializedName: "capacityReservationName",
        type: {
            name: "String"
        }
    }
};
export var cloudServiceName = {
    parameterPath: "cloudServiceName",
    mapper: {
        required: true,
        serializedName: "cloudServiceName",
        type: {
            name: "String"
        }
    }
};
export var commandId = {
    parameterPath: "commandId",
    mapper: {
        required: true,
        serializedName: "commandId",
        type: {
            name: "String"
        }
    }
};
export var diskAccessName = {
    parameterPath: "diskAccessName",
    mapper: {
        required: true,
        serializedName: "diskAccessName",
        type: {
            name: "String"
        }
    }
};
export var diskEncryptionSetName = {
    parameterPath: "diskEncryptionSetName",
    mapper: {
        required: true,
        serializedName: "diskEncryptionSetName",
        type: {
            name: "String"
        }
    }
};
export var diskName = {
    parameterPath: "diskName",
    mapper: {
        required: true,
        serializedName: "diskName",
        type: {
            name: "String"
        }
    }
};
export var diskRestorePointName = {
    parameterPath: "diskRestorePointName",
    mapper: {
        required: true,
        serializedName: "diskRestorePointName",
        type: {
            name: "String"
        }
    }
};
export var edgeZone = {
    parameterPath: "edgeZone",
    mapper: {
        required: true,
        serializedName: "edgeZone",
        type: {
            name: "String"
        }
    }
};
export var expand0 = {
    parameterPath: [
        "options",
        "expand"
    ],
    mapper: {
        serializedName: "$expand",
        type: {
            name: "String"
        }
    }
};
export var expand1 = {
    parameterPath: [
        "options",
        "expand"
    ],
    mapper: {
        serializedName: "$expand",
        type: {
            name: "Enum",
            allowedValues: [
                "instanceView",
                "userData"
            ]
        }
    }
};
export var filter = {
    parameterPath: [
        "options",
        "filter"
    ],
    mapper: {
        serializedName: "$filter",
        type: {
            name: "String"
        }
    }
};
export var forceDeletion = {
    parameterPath: [
        "options",
        "forceDeletion"
    ],
    mapper: {
        serializedName: "forceDeletion",
        type: {
            name: "Boolean"
        }
    }
};
export var galleryApplicationName = {
    parameterPath: "galleryApplicationName",
    mapper: {
        required: true,
        serializedName: "galleryApplicationName",
        type: {
            name: "String"
        }
    }
};
export var galleryApplicationVersionName = {
    parameterPath: "galleryApplicationVersionName",
    mapper: {
        required: true,
        serializedName: "galleryApplicationVersionName",
        type: {
            name: "String"
        }
    }
};
export var galleryImageName = {
    parameterPath: "galleryImageName",
    mapper: {
        required: true,
        serializedName: "galleryImageName",
        type: {
            name: "String"
        }
    }
};
export var galleryImageVersionName = {
    parameterPath: "galleryImageVersionName",
    mapper: {
        required: true,
        serializedName: "galleryImageVersionName",
        type: {
            name: "String"
        }
    }
};
export var galleryName = {
    parameterPath: "galleryName",
    mapper: {
        required: true,
        serializedName: "galleryName",
        type: {
            name: "String"
        }
    }
};
export var galleryUniqueName = {
    parameterPath: "galleryUniqueName",
    mapper: {
        required: true,
        serializedName: "galleryUniqueName",
        type: {
            name: "String"
        }
    }
};
export var hostGroupName = {
    parameterPath: "hostGroupName",
    mapper: {
        required: true,
        serializedName: "hostGroupName",
        type: {
            name: "String"
        }
    }
};
export var hostName = {
    parameterPath: "hostName",
    mapper: {
        required: true,
        serializedName: "hostName",
        type: {
            name: "String"
        }
    }
};
export var imageName = {
    parameterPath: "imageName",
    mapper: {
        required: true,
        serializedName: "imageName",
        type: {
            name: "String"
        }
    }
};
export var includeColocationStatus = {
    parameterPath: [
        "options",
        "includeColocationStatus"
    ],
    mapper: {
        serializedName: "includeColocationStatus",
        type: {
            name: "String"
        }
    }
};
export var instanceId = {
    parameterPath: "instanceId",
    mapper: {
        required: true,
        serializedName: "instanceId",
        type: {
            name: "String"
        }
    }
};
export var location0 = {
    parameterPath: "location",
    mapper: {
        required: true,
        serializedName: "location",
        type: {
            name: "String"
        }
    }
};
export var location1 = {
    parameterPath: "location",
    mapper: {
        required: true,
        serializedName: "location",
        constraints: {
            Pattern: /^[-\w\._]+$/
        },
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var offer = {
    parameterPath: "offer",
    mapper: {
        required: true,
        serializedName: "offer",
        type: {
            name: "String"
        }
    }
};
export var orderby = {
    parameterPath: [
        "options",
        "orderby"
    ],
    mapper: {
        serializedName: "$orderby",
        type: {
            name: "String"
        }
    }
};
export var osFamilyName = {
    parameterPath: "osFamilyName",
    mapper: {
        required: true,
        serializedName: "osFamilyName",
        type: {
            name: "String"
        }
    }
};
export var osVersionName = {
    parameterPath: "osVersionName",
    mapper: {
        required: true,
        serializedName: "osVersionName",
        type: {
            name: "String"
        }
    }
};
export var platformUpdateDomain = {
    parameterPath: "platformUpdateDomain",
    mapper: {
        required: true,
        serializedName: "platformUpdateDomain",
        type: {
            name: "Number"
        }
    }
};
export var privateEndpointConnectionName = {
    parameterPath: "privateEndpointConnectionName",
    mapper: {
        required: true,
        serializedName: "privateEndpointConnectionName",
        type: {
            name: "String"
        }
    }
};
export var proximityPlacementGroupName = {
    parameterPath: "proximityPlacementGroupName",
    mapper: {
        required: true,
        serializedName: "proximityPlacementGroupName",
        type: {
            name: "String"
        }
    }
};
export var publisherName = {
    parameterPath: "publisherName",
    mapper: {
        required: true,
        serializedName: "publisherName",
        type: {
            name: "String"
        }
    }
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        type: {
            name: "String"
        }
    }
};
export var restorePointCollectionName = {
    parameterPath: "restorePointCollectionName",
    mapper: {
        required: true,
        serializedName: "restorePointCollectionName",
        type: {
            name: "String"
        }
    }
};
export var restorePointName = {
    parameterPath: "restorePointName",
    mapper: {
        required: true,
        serializedName: "restorePointName",
        type: {
            name: "String"
        }
    }
};
export var roleInstanceName = {
    parameterPath: "roleInstanceName",
    mapper: {
        required: true,
        serializedName: "roleInstanceName",
        type: {
            name: "String"
        }
    }
};
export var roleName = {
    parameterPath: "roleName",
    mapper: {
        required: true,
        serializedName: "roleName",
        type: {
            name: "String"
        }
    }
};
export var runCommandName = {
    parameterPath: "runCommandName",
    mapper: {
        required: true,
        serializedName: "runCommandName",
        type: {
            name: "String"
        }
    }
};
export var sasUriExpirationTimeInMinutes = {
    parameterPath: [
        "options",
        "sasUriExpirationTimeInMinutes"
    ],
    mapper: {
        serializedName: "sasUriExpirationTimeInMinutes",
        type: {
            name: "Number"
        }
    }
};
export var select = {
    parameterPath: [
        "options",
        "select"
    ],
    mapper: {
        serializedName: "$select",
        type: {
            name: "String"
        }
    }
};
export var sharedTo = {
    parameterPath: [
        "options",
        "sharedTo"
    ],
    mapper: {
        serializedName: "sharedTo",
        type: {
            name: "String"
        }
    }
};
export var skipShutdown = {
    parameterPath: [
        "options",
        "skipShutdown"
    ],
    mapper: {
        serializedName: "skipShutdown",
        defaultValue: false,
        type: {
            name: "Boolean"
        }
    }
};
export var skus = {
    parameterPath: "skus",
    mapper: {
        required: true,
        serializedName: "skus",
        type: {
            name: "String"
        }
    }
};
export var snapshotName = {
    parameterPath: "snapshotName",
    mapper: {
        required: true,
        serializedName: "snapshotName",
        type: {
            name: "String"
        }
    }
};
export var sshPublicKeyName = {
    parameterPath: "sshPublicKeyName",
    mapper: {
        required: true,
        serializedName: "sshPublicKeyName",
        type: {
            name: "String"
        }
    }
};
export var statusOnly = {
    parameterPath: [
        "options",
        "statusOnly"
    ],
    mapper: {
        serializedName: "statusOnly",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
export var top = {
    parameterPath: [
        "options",
        "top"
    ],
    mapper: {
        serializedName: "$top",
        type: {
            name: "Number"
        }
    }
};
export var type = {
    parameterPath: "type",
    mapper: {
        required: true,
        serializedName: "type",
        type: {
            name: "String"
        }
    }
};
export var updateDomain = {
    parameterPath: "updateDomain",
    mapper: {
        required: true,
        serializedName: "updateDomain",
        type: {
            name: "Number"
        }
    }
};
export var version = {
    parameterPath: "version",
    mapper: {
        required: true,
        serializedName: "version",
        type: {
            name: "String"
        }
    }
};
export var virtualMachineScaleSetName = {
    parameterPath: "virtualMachineScaleSetName",
    mapper: {
        required: true,
        serializedName: "virtualMachineScaleSetName",
        type: {
            name: "String"
        }
    }
};
export var vmExtensionName = {
    parameterPath: "vmExtensionName",
    mapper: {
        required: true,
        serializedName: "vmExtensionName",
        type: {
            name: "String"
        }
    }
};
export var vmName = {
    parameterPath: "vmName",
    mapper: {
        required: true,
        serializedName: "vmName",
        type: {
            name: "String"
        }
    }
};
export var vmRestorePointName = {
    parameterPath: "vmRestorePointName",
    mapper: {
        required: true,
        serializedName: "vmRestorePointName",
        type: {
            name: "String"
        }
    }
};
export var vmScaleSetName = {
    parameterPath: "vmScaleSetName",
    mapper: {
        required: true,
        serializedName: "vmScaleSetName",
        type: {
            name: "String"
        }
    }
};
export var vmssExtensionName = {
    parameterPath: "vmssExtensionName",
    mapper: {
        required: true,
        serializedName: "vmssExtensionName",
        type: {
            name: "String"
        }
    }
};
